### DISCUSSION TITLE
3.0.0

### DISCUSSION THREAD

**Discussion by zHaytam** [Blazor-Diagrams/Blazor.Diagrams#341] [Announcements]
Finally, the new documentation website is here!  
Please don't hesitate to create issues for any problems or improvements.  
PS: I suck at design.

### Added

- `AddLabel` method to links to easily create `LinkLabelModel`
- `AddVertex` method to links to easily create `LinkVertexModel`
- `ControlledSize` property to nodes. If `true`, the node will not be registered in the `ResizeObserver` (saves a JS call).
- `autoSize` argument to `SvgGroupModel` constructor

### Changed

- Renamed `Point.Substract` to `Subtract` (duh)
- Avoid rendering link selection helper on dragged link

### Fixed

- `SmoothPathGenerator` not working with `LinkAnchor`
- Mouse overlapping dragged link
- Useless Console Logs from `GroupModel` are now removed
- JS exception in `(un)oberve` methods when the element doesn't exist anymore

Check out the other beta version for a complete CHANGELOG !

<hr /><em>This discussion was created from the release <a href='https://github.com/Blazor-Diagrams/Blazor.Diagrams/releases/tag/3.0.0-release'>3.0.0</a>.</em>

**Comment by ATEEKGIT** [comment]
loved it , so far it is just perfect , thank you so much dude

**Comment by lassemidt** [comment]
New website looks great! Looking forward to try out the 3.0.0 release
